crates/core/tedge_mapper/src/converter.rs

```diff
@@ -21,7 +21,29 @@ pub trait Converter: Send + Sync {
     fn try_convert(&mut self, input: &Message) -> Result<Vec<Message>, Self::Error>;
 
     fn convert(&mut self, input: &Message) -> Vec<Message> {
-        match self.try_convert(input) {
+        let messages_or_err = self.try_convert(input);
+        self.wrap_error(messages_or_err)
+    }
+
+    fn wrap_error(&self, messages_or_err: Result<Vec<Message>, Self::Error>) -> Vec<Message> {
+        match messages_or_err {
+            Ok(messages) => messages,
+            Err(error) => {
+                error!("Mapping error: {}", error);
+                vec![Message::new(
+                    &self.get_mapper_config().errors_topic,
+                    error.to_string(),
+                )]
+            }
+        }
+    }
+
+    fn try_init_messages(&self) -> Result<Vec<Message>, Self::Error> {
```

Comment:
A `Convertor` trait meant for conversion specific logic having a method like this that generates messages out of thin air looks a bit weird. Wondering if this function really belongs in this trait or rather be a part of the initialization logic of the mapper elsewhere.

Reply:
Maybe the name `Converter` is misleading, but this trait responsibilities make sense. These are related to the messages to be sent to a given cloud. To translate a measurements, create a child device ... or declare that an operation is supported. The mapper main itself is then responsible of the transmission of these messages and can be cloud independent.